src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -93,21 +97,75 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
         }
     }
 
+    private bool TryGetMetadataPath(string key, [NotNullWhen(true)]out string? path)
+    {
+        var entryAssembly = Assembly.GetEntryAssembly()!; // Null suppression because it can't return null in this context.
```

Comment:
You should use the assembly in `DistributedApplicationOptions`. Move the processing where we process all of the other attributes.